src/aura/CRLP_FilterGroup/CRLP_FilterGroupHelper.js

```diff
@@ -536,6 +538,18 @@
             return validSoFar && filterRuleCmp.get("v.validity").valid;
         }, true);
 
+        //check for specific API name duplicates instead of the full row to avoid issues with the ID not matching
+        for (var i = 0; i < filterRuleList.length; i++) {
+            if (i !== filterRule.index
+                && filterRuleList[i].objectName === filterRule.objectName
+                && filterRuleList[i].fieldName === filterRule.fieldName
+                && filterRuleList[i].operatorName === filterRule.operatorName
+                && filterRuleList[i].constantName === filterRule.constantName) {
+                cmp.set("v.filterRuleError", cmp.get("v.labels.filterRuleDuplicate"));
+                return canSave = false;
```

Comment:
Is there a need to use `return canSave = false;` versus just `return false;` in this function? Seems like `canSave` is a local var only.